docs/how-to/service-dependencies.md

```diff
@@ -1,11 +1,270 @@
+("inactive"), but the ``backend`` and ``frontend`` services starting
+successfully ("active"):
+
+```{terminal}
+   :input: pebble run
+
+2024-06-28T02:20:03.337Z [pebble] Started daemon.
+[backend database frontend]
+2024-06-28T02:20:03.343Z [pebble] POST /v1/services 2.763792ms 202
+2024-06-28T02:20:03.346Z [pebble] Service "backend" starting: python3 -m http.server 8081
+2024-06-28T02:20:03.346Z [pebble] Service "database" starting: python3 -m http.server 3306
+2024-06-28T02:20:03.347Z [pebble] Service "frontend" starting: python3 -m http.server 8080
+2024-06-28T02:20:03.396Z [pebble] Change 1 task (Start service "database") failed: cannot start service: exited quickly with code 1
+2024-06-28T02:20:04.353Z [pebble] Started default services with change 1.
+```
+
+```{terminal}
+   :input: pebble services
+   
+Service   Startup  Current   Since
+backend   enabled  active    today at 10:20 CST
+database  enabled  inactive  -
+frontend  enabled  active    today at 10:20 CST
+```
+
+We can configure the layer so that Pebble only starts a given service if all
+other services it is dependent on are running successfully to avoid consuming
+resources unnecessarily.
+
+## Define service dependencies
+
+To create service dependencies in Pebble, use the ``requires`` key with
+``before`` / ``after`` in the
+[service definition](../reference/layer-specification.md).
+
+### Specify dependent services
+
+To specify one or more services that a given service requires to run
+successfully, use the ``require`` key.
+
+```{code-block} yaml
+   :emphasize-lines: 6, 7, 12, 13
+
+services:
+  frontend:
+    override: replace
+    command: python3 -m http.server 8080
+    startup: enabled
+    requires:
+      - backend
+  backend:
+    override: replace
+    command: python3 -m http.server 8081
+    startup: enabled
+    requires:
+      - database
+  database:
+    override: replace
+    command: python3 -m http.server 3306
+    startup: enabled
+```
+
+In the layer configuration above, the ``backend`` service is dependent on the
+``database`` service, and the ``frontend`` service is dependent on the
+``backend`` service.
+
+### Specify start order for dependent services
+
+To specify the order in which one or more dependent services must start
+successfully, relative to a given service, use the ``before`` or ``after`` keys.
+
+```{code-block} yaml
+   :emphasize-lines: 8, 9, 16, 17
+
+services:
+  frontend:
+    override: replace
+    command: python3 -m http.server 8080
+    startup: enabled
+    requires:
+      - backend
+    after:
+      - backend
+  backend:
+    override: replace
+    command: python3 -m http.server 8081
+    startup: enabled
+    requires:
+      - database
+    after:
+      - database
+  database:
+    override: replace
+    command: python3 -m http.server 3306
+    startup: enabled
+```
+
+In the updated layer above, the ``backend`` service requires the ``database``
```

Comment:
I think this paragraph should mention the services in the order they appear above, so start with frontend, then proceed to backend (rather than the other way around).

Reply:
Will update it @benhoyt .